snapcraft/internal/repo.py

```diff
@@ -299,7 +299,10 @@ def _get(self, apt_cache, package_names):
                 logger.debug(
                     'Marking {!r} (and its dependencies) to be fetched'.format(
                         name))
-                apt_cache[name].mark_install()
+                version = None
+                name_arch, version = _get_pkg_name_parts(name)
+                _set_candidate(apt_cache[name_arch], version)
```

Comment:
Heh, yeah I thought it was the other `get` for a minute, my apologies. To reverse my statement: this function is actually quite large. Splitting it up probably makes sense.